dotnetv3/EventBridge/Actions/EventBridgeWrapper.cs

```diff
@@ -0,0 +1,452 @@
+// SPDX-License-Identifier:  Apache-2.0
+
+using System.Net;
+using System.Text.Json;
+using Amazon.EventBridge;
+using Amazon.EventBridge.Model;
+using Microsoft.Extensions.Logging;
+
+namespace EventBridgeActions;
+
+// snippet-start:[EventBridge.dotnetv3.EventBridgeWrapper]
+
+/// <summary>
+/// Wrapper for Amazon EventBridge operations.
+/// </summary>
+public class EventBridgeWrapper
+{
+    private readonly IAmazonEventBridge _amazonEventBridge;
+    private readonly ILogger<EventBridgeWrapper> _logger;
+
+    /// <summary>
+    /// Constructor for the EventBridge wrapper.
+    /// </summary>
+    /// <param name="amazonEventBridge">The injected EventBridge client.</param>
+    /// <param name="logger">The injected logger for the wrapper.</param>
+    public EventBridgeWrapper(IAmazonEventBridge amazonEventBridge, ILogger<EventBridgeWrapper> logger)
+
+    {
+        _amazonEventBridge = amazonEventBridge;
+        _logger = logger;
+    }
+
+    // snippet-start:[EventBridge.dotnetv3.DescribeRule]
+    /// <summary>
+    /// Get the state for a rule by the rule name.
+    /// </summary>
+    /// <param name="ruleName">The name of the rule.</param>
+    /// <param name="eventBusName">The optional name of the event bus. If empty, uses the default event bus.</param>
+    /// <returns>The state of the rule.</returns>
+    public async Task<RuleState> GetRuleStateByRuleName(string ruleName, string? eventBusName = null)
+    {
+        var ruleResponse = await _amazonEventBridge.DescribeRuleAsync(
+            new DescribeRuleRequest()
+            {
+                Name = ruleName,
+                EventBusName = eventBusName
+            });
+        return ruleResponse.State;
+    }
+    // snippet-end:[EventBridge.dotnetv3.DescribeRule]
+
+    // snippet-start:[EventBridge.dotnetv3.EnableRule]
+    /// <summary>
+    /// Enable a particular rule on an event bus.
+    /// </summary>
+    /// <param name="ruleName">The name of the rule.</param>
+    /// <returns>True if successful.</returns>
+    public async Task<bool> EnableRuleByName(string ruleName)
+    {
+        var ruleResponse = await _amazonEventBridge.EnableRuleAsync(
+            new EnableRuleRequest()
+            {
+                Name = ruleName
+            });
+        return ruleResponse.HttpStatusCode == HttpStatusCode.OK;
+    }
+    // snippet-end:[EventBridge.dotnetv3.EnableRule]
+
+    // snippet-start:[EventBridge.dotnetv3.DisableRule]
+    /// <summary>
+    /// Disable a particular rule on an event bus.
+    /// </summary
+    /// <param name="ruleName">The name of the rule.</param>
+    /// <returns>True if successful.</returns>
+    public async Task<bool> DisableRuleByName(string ruleName)
+    {
+        var ruleResponse = await _amazonEventBridge.DisableRuleAsync(
+            new DisableRuleRequest()
+            {
+                Name = ruleName,
+                //EventBusName = eventBusName
```

Comment:
I'd remove this unless there's a reason to keep it.

Reply:
Fixed.